File: src/main/java/ipp/PdfPrintJob.java
package ipp;

import static de.gmuth.ipp.client.IppTemplateAttributes.documentFormat;
import static de.gmuth.ipp.client.IppTemplateAttributes.jobName;

import de.gmuth.ipp.client.IppColorMode;
import de.gmuth.ipp.client.IppConfig;
import de.gmuth.ipp.client.IppJob;
import de.gmuth.ipp.client.IppPrinter;
import de.gmuth.ipp.client.IppSides;
import java.io.File;

public class PdfPrintJob {

  public static void main(String[] args) {

    IppConfig ippConfig = new IppConfig();
    ippConfig.setChunkedStreamingMode(false);
    //HttpURLConnectionClient.Companion.getLog().setLogLevel(LogLevel.DEBUG);

    IppPrinter ippPrinter = new IppPrinter("ipp://localhost:8632/printers/laser", ippConfig);
    ippPrinter.logDetails();

    File file = new File("test-docs/A4-ten-pages.pdf");
    IppJob job = ippPrinter.printJob(
      file,
      documentFormat("application/pdf"),
      jobName(file.getName()),
      IppColorMode.Monochrome,
      IppSides.TwoSidedLongEdge
    );
    job.waitForTermination();
    job.logDetails();
  }

}

File: src/main/kotlin/ipp/PdfPrintJob.kt
package ipp

import de.gmuth.ipp.client.IppColorMode
import de.gmuth.ipp.client.IppPrinter
import de.gmuth.ipp.client.IppTemplateAttributes.documentFormat
import de.gmuth.ipp.client.IppTemplateAttributes.jobName
import de.gmuth.ipp.client.IppTemplateAttributes.pageRanges
import de.gmuth.log.Slf4jLogging
import java.io.File

fun main() {
    Slf4jLogging.configure()
    val ippPrinter = IppPrinter("ipp://localhost:8632/printers/laser")
    ippPrinter.logDetails()

    val file = File("test-docs/A4-ten-pages.pdf")
    val job = ippPrinter.printJob(
            file,
            documentFormat("application/pdf"),
            jobName(file.name),
            pageRanges(2..5),
            IppColorMode.Monochrome
    )
    job.logDetails()

    job.waitForTermination()
    job.logDetails()
}